### DISCUSSION TITLE
Dungeons?

### DISCUSSION THREAD

**Discussion by xnacly** [HypixelDev/PublicAPI#277] [Ideas]
I would like to see dungeonskill and class based skills in this api.

**Comment by doej1367** [comment]
Agreed.
To clarify your point:

**[Suggestion] Add Dungeonering Skill XP and Class Level XP**

Please considder adding the following attributes to the API:
+ experience_skill_catacombs
+ experience_skill_healer
+ experience_skill_mage
+ experience_skill_berserk
+ experience_skill_archer
+ experience_skill_tank

**Comment by xnacly** [comment]
thanks and yes i didnt phrase myself the right way, thanks for helping me with this 😄

**Comment by shadowtime2000** [comment]
@doej1367 I thought that there was no catacombs level and instead catacombs level is with each class inside the catacombs dungeon.

**Comment by doej1367** [comment]
@shadowtime2000 I am pretty sure there is, but it is odd indeed that I have the same Catacombs level XP as I have tank Class level XP (I only played tank so far).
Makes you wonder if that is just how it is (you get the same amount of both)
or if they are indeed the same thing...
But why would there need to be a Catacombs level if they are the same?

**Comment by shadowtime2000** [comment]
@doej1367 I believe there is no Catacombs level and if you have played as tank so far then that is your Catacombs level with tank. If you play, say archer, then you would have 0 Catacomb XP at the start. For every class, there is XP for every dungeon.

**Comment by b1n4ryj4n** [comment]
Wrong there is a catacombs level you can see it iff you open the dungeoneering skill and click the button right next to the close button named "Dungeons", so each comming soon dungeon has its own progress lvl. And there are not the same as class levels.

**Comment by doej1367** [comment]
> they are not the same as class levels


Correct. Since I have a Thesis Accessory the two XP levels aren't the same anymore.

**Comment by shadowtime2000** [comment]
Another suggestion:
# [Suggestion] Add Essence Tracking
I think these attributes to the API would be useful:
* essence_undead
* essence_dragon
* essence_gold
* essence_diamond
* essence_wither
* essence_spider

**Comment by shadowtime2000** [comment]
I am assuming this would be closed once the documentation is added for the [0.9.1 patch](https://hypixel.net/threads/skyblock-patch-0-9-1-catacombs-floor-5-and-general-changes.3349106/) API changes

**Comment by builder-247** [comment]
Seems essence upgrades haven't been added yet https://github.com/HypixelDatabase/HypixelTracking/compare/6c75068a03...507a86c36e, this would be essential for correctly calculating player stats such as health, strength etc.
